### ISSUE TITLE
yara_mapper: Assertion failed : length(me.cigar) <= length(me.cigarSet[matchId])

### ISSUE THREAD

**Issue by jhu99** [seqan/seqan#2513]
Dear SeqAn developer,

When I ran Yara_mapper on our data. One error stop it on two of our sample as follows:
```
./tutorial/seqan/apps/yara/mapper_aligner.h:243 Assertion failed : length(me.cigar) <= length(me.cigarSet[matchId]) should be true but was 0 (CIGAR error.)
stack trace:
  0          [0x53c15e]  yara_mapper()
  1          [0xa0481d]  yara_mapper()
  2    [0x7fc22e2a7ed6]  /home/huj/.local/bin/gcc9/lib64/libgomp.so.1(+0x19ed6)
  3    [0x7fc22e8f51cf]  /lib64/libpthread.so.0(+0x81cf)
  4    [0x7fc22d58edd3]  clone + 0x43
/var/spool/slurm/d/job1766663/slurm_script: line 34: 3153523 Aborted                 (core dumped) yara_mapper -t 30 -rg S${L2ID}L2 -sm omit -o ./results/phase2/S${L2ID}_L002.yara.bam ./data/GRCh38/yara/hg38.index ./results/phase2/S${L2ID}_L002_R1_001_val_1.fq.gz ./results/phase2/S${L2ID}_L002_R2_001_val_2.fq.gz


./tutorial/seqan/apps/yara/mapper_aligner.h:243 Assertion failed : length(me.cigar) <= length(me.cigarSet[matchId]) should be true but was 0 (CIGAR error.)
stack trace:
  0          [0x53c15e]  yara_mapper()
  1          [0xa0481d]  yara_mapper()
  2    [0x7f767ba0bed6]  /home/huj/.local/bin/gcc9/lib64/libgomp.so.1(+0x19ed6)
  3    [0x7f767c0591cf]  /lib64/libpthread.so.0(+0x81cf)
  4    [0x7f767acf2dd3]  clone + 0x43
/var/spool/slurm/d/job1766655/slurm_script: line 34: 930794 Aborted                 (core dumped) yara_mapper -t 30 -rg S${SLURM_ARRAY_TASK_ID}L1 -sm omit -o ./results/phase2/S${SLURM_ARRAY_TASK_ID}_L001.yara.bam ./data/GRCh38/yara/hg38.index ./results/phase2/S${SLURM_ARRAY_TASK_ID}_L001_R1_001_val_1.fq.gz ./results/phase2/S${SLURM_ARRAY_TASK_ID}_L001_R2_001_val_2.fq.gz
```

Could you give me some tips how to fix this issue?

Best
Jialu

**Comment by smehringer** [comment]
Hi @jhu99,

thanks for reaching out and your interest in our tools!

We need some more information to investigate your issue. 

First, Which Yara Mapper version (a.k.a seqan version) do you use? (`yara_mapper --version`)

Is it possible that you provide us with the data that fails, s.t. we can reproduce your error? I see that you are mapping against hg38, that of course we have available, but we would need the fastq file that errors. It would be enough (and even be beneficial) to get the single read (pair) that errors?

Best,
Svenja

**Comment by jhu99** [comment]
Hi Svenja,

Thanks for your fast response.

Please find the version information as follows:
yara_mapper version: 1.0.3 [77c2180]
SeqAn version: 2.5.0

I'am preparing the fastq files, and will send it to you.

Best,
Jialu

**Comment by eseiler** [comment]
We resolved this problem, hence closing the issue.

There were two problems:

#### Outdated `cutadapt`

The `yara_mapper` error was resolved by updating `cutadapt` from `v2.6` to `v4.7`.

#### BAM files were missing the `@SQ` headers

`seqan3` was unable to read the resulting BAM files.
This was due to the BAM files missing `@SQ` headers, which resulted in `seqan3::sam_file_input` encountering unknown references.

This is fixed in the `main` branch by #2515.

Fixing the erroneous BAM files can be done by:
* Running the updated `yara_mapper` again
* Using `samtools` to fix the corrupted files:
  ```bash
  samtools view --with-header --bam --threads 32 corrupted.yara.bam > fixed.yara.bam 
  ```
